File: ecos-records/src/main/java/ru/citeck/ecos/predicate/comparator/ValueComparator.java
package ru.citeck.ecos.predicate.comparator;

public interface ValueComparator {

    boolean isEquals(Object value0, Object value1);

    boolean isContains(Object value, Object subValue);

    boolean isIn(Object value, Object inValue);

    boolean isGreaterThan(Object value, Object thanValue, boolean inclusive);

    boolean isLessThan(Object value, Object thanValue, boolean inclusive);

    boolean isLike(Object value, Object likeValue);

    boolean isEmpty(Object value);
}

File: ecos-records/src/main/java/ru/citeck/ecos/records2/predicate/comparator/ValueComparator.kt
package ru.citeck.ecos.records2.predicate.comparator

import ru.citeck.ecos.commons.data.DataValue

interface ValueComparator {

    fun isEquals(value0: DataValue, value1: DataValue): Boolean

    fun isContains(value: DataValue, subValue: DataValue): Boolean

    fun isIn(value: DataValue, inValue: DataValue): Boolean

    fun isGreaterThan(value: DataValue, thanValue: DataValue, inclusive: Boolean): Boolean

    fun isLessThan(value: DataValue, thanValue: DataValue, inclusive: Boolean): Boolean

    fun isLike(value: DataValue, likeValue: DataValue): Boolean

    fun isEmpty(value: DataValue): Boolean
}
